src/Detector.php

```diff
@@ -80,16 +80,22 @@ public function determineDimensions(\SplFileObject $file): array
             if (false === $info) {
                 throw new \InvalidArgumentException('Failed to get the image size. Given path is ' . $path . '.');
             }
+        }
```

Comment:
This change has removed the `return [$info[0], $info[1]];`. This should be put back, otherwise the code will have established the dimensions already and still continue with copying the entire file to memory.

Reply:
@Zegnat. Thank you for your reply. You're right and this will be put back.

The code coverage is decreased and I will add more tests to increase the code coverage!